### ISSUE TITLE
Parsing exception on PEP 586 Literal (Using typing-extensions)

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#769]
In GitLab by @jonb on Sep 17, 2020, 15:55

Please read this brief portion of documentation before going any further: http://flake8.pycqa.org/en/latest/internal/contributing.html#filing-a-bug

*Please describe how you installed Flake8*

```
Using pygradle.
```

**Note**: Some *nix distributions patch Flake8 arbitrarily to accommodate incompatible software versions. If you're on one of those distributions, your issue may be closed and you will be asked to open an issue with your distribution package maintainers instead.

*Please provide the exact, unmodified output of `flake8 --bug-report`*

```
$ flake8 --bug-report
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.7",
    "system": "Linux"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "flake8-annotations-complexity",
      "version": "0.0.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-bugbear",
      "version": "20.1.4"
    },
    {
      "is_local": false,
      "plugin": "flake8-pie",
      "version": "0.5.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_builtins",
      "version": "1.4.1"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.6.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.2.0"
    }
  ],
  "version": "3.8.3"
}
$
```

*Please describe the problem or feature*

When using a PEP 586 Literal type, a parsing exception is thrown. A small reproducer:

```
from typing_extensions import Literal
foo: Literal['*'] = '*'
```

See Also:

* "Declaring literal variables" in https://mypy.readthedocs.io/en/stable/literal_types.html
* Various examples under "Intelligent indexing of structured data" in https://www.python.org/dev/peps/pep-0586/

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

What happened: the exception noted below was printed.

What I expect: no exception.

```$ cat blah.py
from typing_extensions import Literal
foo: Literal['*'] = '*'
$ flake8 blah.py
Traceback (most recent call last):
  File "/XXX/environments/development-venv/bin/flake8", line 11, in <module>
    sys.exit(main())
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/main/cli.py", line 22, in main
    app.run(argv)
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/main/application.py", line 360, in run
    self._run(argv)
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/main/application.py", line 348, in _run
    self.run_checks()
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/main/application.py", line 262, in run_checks
    self.file_checker_manager.run()
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/checker.py", line 325, in run
    self.run_serial()
  File "/XXX/development-venv/lib/python3.7/site-packages/flake8/checker.py", line 309, in run_serial
    checker.run_checks()
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/checker.py", line 589, in run_checks
    self.run_ast_checks()
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8/checker.py", line 496, in run_ast_checks
    for (line_number, offset, text, _) in runner:
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8_annotations_complexity/checker.py", line 25, in run
    self.max_annotations_complexity,
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8_annotations_complexity/ast_helpres.py", line 28, in validate_annotations_in_ast_node
    complexity = get_annotation_compexity(annotation)
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8_annotations_complexity/ast_helpres.py", line 9, in get_annotation_compexity
    return 1 + get_annotation_compexity(annotation_node.slice.value)  # type: ignore
  File "/XXX/environments/development-venv/lib/python3.7/site-packages/flake8_annotations_complexity/ast_helpres.py", line 7, in get_annotation_compexity
    annotation_node = ast.parse(annotation_node.s).body[0].value  # type: ignore
  File "/XXX/python/3.7.7/lib/python3.7/ast.py", line 35, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 1
    *
    ^
SyntaxError: invalid syntax
$```

**Comment by asottile** [comment]
In GitLab by @asottile on Sep 17, 2020, 15:59

as you can see from the trace this isn't a problem with flake8, please open an issue on the project that's causing the trace

**Comment by asottile** [comment]
In GitLab by @jonb on Sep 17, 2020, 16:00

Yep... I just realized it's https://github.com/best-doctor/flake8-annotations-complexity -- Sorry!

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Sep 17, 2020, 16:00

How're you so speedy @asottile ? I was just trying to save you from having to close this one too

**Comment by asottile** [comment]
In GitLab by @jonb on Sep 17, 2020, 16:03

If anyone searching for this ticket comes across this -- this issue appears to be fixed in flake8-annotations-complexity 0.0.5.
